src/In.ProjectEKA.HipService/Discovery/Filter.cs

```diff
@@ -56,8 +56,12 @@ private static IEnumerable<IdentifierExt> From(DiscoveryRequest request)
         public IEnumerable<PatientEnquiryRepresentation> Do(IEnumerable<Patient> patients,
             DiscoveryRequest request)
         {
-            return patients
-                .AsEnumerable()
+            var unverifiedIdMatches = (from unverifiedIdentifier in request.Patient?.UnverifiedIdentifiers ?? new List<Identifier>() 
+                where unverifiedIdentifier.Type == IdentifierType.MR 
+                    from patient in patients.Where(p => p.Identifier?.Equals(unverifiedIdentifier.Value) ?? false) select patient).ToList();
+            var matchedBy = unverifiedIdMatches.Count > 0 ? new List<string>() { "MR" } : new List<string>();  
+            var filteredPatients = unverifiedIdMatches.Count > 0 ? unverifiedIdMatches : patients; 
+            return filteredPatients
```

Comment:
why can't you run it through rankers? I'm pretty sure it will work, as it's unverified identifier to filter out one patient from list of patients.